dbms/src/Flash/Planner/Plans/PhysicalCTESource.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Coprocessor/ChunkCodec.h>
+#include <Flash/Coprocessor/DAGContext.h>
+#include <Flash/Pipeline/Exec/PipelineExecBuilder.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Flash/Planner/Plans/PhysicalCTESource.h>
+#include <Interpreters/Context.h>
+#include <Operators/CTEReader.h>
+#include <Operators/CTESource.h>
+
+#include <memory>
+#include <string>
+
+namespace DB
+{
+PhysicalPlanNodePtr PhysicalCTESource::build(
+    const Context & /*context*/,
+    const String & executor_id,
+    const LoggerPtr & log,
+    const FineGrainedShuffle & fine_grained_shuffle,
+    const tipb::CTESource & cte_source)
+{
+    DAGSchema dag_schema;
+    for (int i = 0; i < cte_source.field_types_size(); ++i)
+    {
+        String name = genNameForCTESource(i);
+        TiDB::ColumnInfo info = TiDB::fieldTypeToColumnInfo(cte_source.field_types(i));
+        dag_schema.emplace_back(std::move(name), std::move(info));
+    }
+
+    NamesAndTypes schema = toNamesAndTypes(dag_schema);
+    return std::make_shared<PhysicalCTESource>(
+        executor_id,
+        schema,
+        fine_grained_shuffle,
+        log->identifier(),
+        Block(schema),
+        cte_source.cte_id(),
+        cte_source.cte_sink_num(),
+        cte_source.cte_source_num());
+}
+
+void PhysicalCTESource::buildPipelineExecGroupImpl(
```

Comment:
Is it expected that `buildBlockInputStreamImpl()` didn't implement

Reply:
> Is it expected that `buildBlockInputStreamImpl()` didn't implement

yes